### PR TITLE
feat(ethereum-api): <- adds version targeting solc 0.8.x

### PR DESCRIPTION
**Pull request by gskapka**
...per title.

Also adds:

 - Hardhat support
 - Compilation tests
 - CI
 - Dir refactoring to clean up repo
 - Updated `README.md` to show latest way to import API into remix.org

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,15 +8,34 @@ In Solidity it is as simple as inheriting the __`usingProvable`__ contract that
 
 This will provide your contract with functions like __`provable_query(...)`__, which make it trivial for you to leverage our oracle technology straight away.
 
-If you're using the __[Remix IDE](http://remix.ethereum.org)__ it's even easier still - simply import __Provable__ into your contract like so:
+If you're using the __[Remix IDE](http://remix.ethereum.org)__ it's even easier still - simply import the latest version of __Provable__ into your contract like so:
 
 ```solidity
 
-import "github.com/oraclize/ethereum-api/provableAPI.sol";
+import "github.com/provable-things/ethereum-api/provableAPI.sol";
```
Comment (oliviera9) [inline README.md:15]: We may add a note that, if older versions are needed, one would need to import from the specific folders.

Comment (gskapka) [reply README.md:15]: Good point. Doing it now...

Comment (gskapka) [reply README.md:15]: Ah wait, it's already done :rofl:

**Review by oliviera9 — APPROVED** [review]
LGTM

**Pull request merged**